hawkular-alerts-rest/src/main/java/org/hawkular/alerts/rest/TriggersHandler.java

```diff
@@ -451,6 +457,9 @@ public Response updateTrigger(
 
         } catch (Exception e) {
             log.debug(e.getMessage(), e);
+            if (e.getCause() != null && e.getCause() instanceof IllegalArgumentException) {
+                return ResponseUtil.badRequest("Bad arguments: " + e.getMessage());
+            }
```

Comment:
It's not tedious. I think it's harmless to extend same discrimination in the catch to other areas.
I will send a fix.